### PR TITLE
feat(dnd) : #MAG-486 dnd impl vertical and horizontal

### PR DESCRIPTION
**Pull request by alicedraillard**
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/board-card/BoardCard.tsx [hunk 1]
```diff
@@ -65,7 +65,13 @@ export const BoardCard: FC<BoardCardProps> = ({
     setNodeRef,
     transform,
     transition,
-  } = useSortable({ id: card.id });
+  } = useSortable({
+    id: card.id,
+    data: {
+      type: "card",
```
Comment (niko-amosse) [inline frontend/src/components/board-card/BoardCard.tsx:71]: const ?

File: frontend/src/components/board-card/BoardCard.tsx [hunk 2]
```diff
@@ -83,6 +89,7 @@ export const BoardCard: FC<BoardCardProps> = ({
   return (
     <StyledCard
       data-dropdown-open={isOpen ? "true" : "false"}
+      data-type="card"
```
Comment (niko-amosse) [inline frontend/src/components/board-card/BoardCard.tsx:92]: idem ?

File: frontend/src/components/dnd-section/DndSection.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    attributes,
+    listeners,
+    setNodeRef,
+    transform,
+    transition,
+    isDragging,
+  } = useSortable({
+    id,
+    data: {
+      type: "section",
```
Comment (niko-amosse) [inline frontend/src/components/dnd-section/DndSection.tsx:27]: idem ?

File: frontend/src/components/dnd-section/DndSection.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  const style = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+  };
+
+  return (
+    <SectionWrapper
+      ref={setNodeRef}
+      style={style}
+      data-type="section"
```
Comment (niko-amosse) [inline frontend/src/components/dnd-section/DndSection.tsx:44]: idem ?

**Review by niko-amosse — APPROVED** [review]

File: frontend/src/components/dnd-section/style.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+}) => {
+  switch (sectionType) {
+    case "vertical":
+      return {
+        boxSizing: "border-box",
+        display: "flex",
+        flexDirection: "column",
+        gap: "2rem",
+        padding: "1rem 0.1rem 1rem 0",
+        minWidth: "420px",
```
Comment (FlorentMr) [inline frontend/src/components/dnd-section/style.ts:29]: PX

**Review by FlorentMr — APPROVED** [review]

**Pull request merged**